airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumer.kt

```diff
@@ -160,18 +159,11 @@ constructor(
         bufferManager.close()
 
         val streamSyncSummaries =
-            streamNames
-                .stream()
-                .collect(
-                    Collectors.toMap(
-                        { streamDescriptor: StreamDescriptor -> streamDescriptor },
-                        { streamDescriptor: StreamDescriptor ->
-                            StreamSyncSummary(
-                                Optional.of(getRecordCounter(streamDescriptor).get()),
-                            )
-                        },
-                    ),
+            streamNames.associateWith { streamDescriptor: StreamDescriptor ->
```

Comment:
ooh, TIL associateWith

(i'm immediately switching to `associate` [here](https://github.com/airbytehq/airbyte/pull/38121/files#diff-1cf39e4350409debcfd4aa5ebf18ac80a12d0ecce9c0cfd0d727b8ff32397361R133) but it's still cool!)